micrometer-spring-legacy/src/main/java/io/micrometer/spring/autoconfigure/web/jetty/JettyMetricsAutoConfiguration.java

```diff
@@ -46,11 +53,26 @@
         "javax.servlet.Servlet", "org.eclipse.jetty.server.Server", "org.eclipse.jetty.util.Loader",
         "org.eclipse.jetty.webapp.WebAppContext" })
 @Conditional(JettyMetricsAutoConfiguration.JettyMetricsAutoConfigurationConditionalOnBeans.class)
+@ConditionalOnBean(AnnotationConfigEmbeddedWebApplicationContext.class)
```

Comment:
Sometimes I am surprised by what's available as a bean. This is one of those times. Hopefully we can come up with a more robust way of dealing with this, but I'm not yet sure what it should be. A custom condition might make the intent clearer if nothing else.

Reply:
@wilkinsona Thanks for the feedback! I changed to use a custom condition to check whether it's an embedded web application in a32aa8a.